event_model/documents/stream_datum.py

```diff
@@ -55,3 +41,24 @@ class StreamDatum(TypedDict):
             "formatting being '<stream_resource>/<stream_name>/<block_id>",
         ),
     ]
+    data_keys: Annotated[
+        List[str],
+        Field(
+            description="A list to show which data_keys of the "
+            "Descriptor are being streamed"
+        ),
+    ]
+    seq_nums: Annotated[
+        Dict[str, int],
```

Comment:
Could we make this another TypedDict rather than a generic Dict please?
```suggestion
        Range,
```
Where `Range` is:
```python
class Range(TypedDict):
    """The parameters required to describe a sequence of incrementing integers"""
    start: Annotated[
        int,
        Field(description="First number in the range"),
    ]
    stop: Annotated[
        int,
        Field(description="Last number in the range is less than this number"),
    ]
```